suites/all_vartime.go

```diff
@@ -5,11 +5,18 @@ package suites
 import (
 	"github.com/dedis/kyber/group/curve25519"
 	"github.com/dedis/kyber/group/nist"
+	"github.com/dedis/kyber/pairing/bn256"
 )
 
 func init() {
 	register(curve25519.NewBlakeSHA256Curve25519(false))
 	register(curve25519.NewBlakeSHA256Curve25519(true))
 	register(nist.NewBlakeSHA256P256())
 	register(nist.NewBlakeSHA256QR512())
+	register(bn256.NewSuiteG1())
+	register(bn256.NewSuiteG2())
+	register(bn256.NewSuiteGT())
+	register(bn256.NewSuite().G1().(Suite))
```

Comment:
What's the point of registering twice every group ?

Reply:
one registration is the 'clean' `G1`, another registration is the `bn256`-suite with a `G1` `group` in it. Two examples:
```
s := suites.MustFind("bn256.g1")
// This does not work
private, public := bls.NewKeyPair(s, random.New())
```
```
s := suites.MustFind("combined:bn256.g1")
// This works
private, public := bls.NewKeyPair(s, random.New())
```

Depending on the use case you can have one or the other. Even though, only the `NewSuiteG2` might make sense. @JoaoAndreSa is using `G1` as a suite for the cothority - but that needs perhaps to be changed.

Reply:
What's the reason the first example does not work ? I thought the whole point of making G1 into a suit was to make it work as such ... ><

Reply:
The first example creates a `Suite` with only a `Group` of `G1` - this is of course not compatible with `pairing.Suite`.

Reply:
Anyway - most of this PR is due to dedis/onet#432...

Reply:
So ok i see the problem. So that's one solution. 
Another solution could be to juste make PairingSuite returns G1 and G2 and GT as Suite ?
But that''ll change too much code and is uglier because it even puts more thing into the Suite.. 
Ok for solution 1 ;)
aaarrgggggg I am starting to be allergic to these suites thing !!

Reply:
As I wrote above, it's mostly due to dedis/onet#432 and how people want to use it in onet/cothority. Thinking it through, most probably the best place to add it then is onet or cothority. Now the changes David needed to do is the part in Marshaling, that is already merged. I'll have a look if I can take it to `cothority/suite.go`.